src/main/java/com/squareup/kotlinpoet/CodeWriter.kt

```diff
@@ -445,7 +432,6 @@ internal class CodeWriter constructor(
       out.append(line)
       trailingNewline = false
     }
-    return this
```

Comment:
Is there a benefit to returning self in Kotlin? Is it idiomatic?

Reply:
There is little benefit because of `apply { }` blocks. I was actually considering migrating to `CodeWriter` uses `apply` as a follow-up.

Reply:
Just tried it. It works in some areas and doesn't in others.

Things like
```kotlin
if (index > 0) codeWriter.emit(",").emitWrappingSpace()
```
are forced to become
```kotlin
if (index > 0) {
  codeWriter.apply {
    emit(", ")
    emitWrappingSpace()
  }
}
```

Reply:
I like the first one!